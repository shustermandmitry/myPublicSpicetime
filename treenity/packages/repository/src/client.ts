export * from './client/access-token';
export * from './client/create-feathers-client';
export * from './client/simple-client';
export * from './utils/cookies';
export * from './client/socket-custom-methods';
export * from './mods/client';
export * from './client/parse-jwt';
export * from './utils/email-parse';
export { makeServiceActions } from '@/mods/entity/service-entity-actions';
