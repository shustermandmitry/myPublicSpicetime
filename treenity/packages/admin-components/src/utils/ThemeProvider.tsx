/*
 * Copyright (c) 2024. Treenity Inc.
 */

import { setCookie, useCookie } from '@treenity/repository/client';
import { TreeNode, useTree } from '@treenity/tree-api';
import { useStorage } from '@treenity/ui-kit/hooks';
import { Theme, ThemeEntity, ThemeProvider } from '@treenity/ui-kit/theme';
import React, { FC, PropsWithChildren } from 'react';
import useSWR from 'swr';

const defaultThemeName = 'light';

function useLoadTheme(name: string): Theme | undefined {
  const treeApi = useTree();
  const { data: themeNode } = treeApi.useNode(`/sys/themes/${name}`);
  const { data } = useSWR(`${name}-theme` + (themeNode as any)?.$id, () =>
    (themeNode as TreeNode)?.get(ThemeEntity),
  );
  return data;
}

const ThemeProviderLoader: FC<PropsWithChildren> = ({ children }) => {
  const lightTheme = useLoadTheme('light');
  const darkTheme = useLoadTheme('dark');

  const cookies = useCookie();
  const [storedThemeName, _setThemeName] = useStorage<string>(
    'theme',
    cookies.theme || defaultThemeName,
  );
  const setThemeName = (theme: string) => {
    setCookie('theme', theme, 365 * 24 * 60 * 60);
    _setThemeName(theme);
  };

  if (!lightTheme || !darkTheme) return null;

  return (
    <ThemeProvider
      light={lightTheme}
      dark={darkTheme}
      children={children}
      storedThemeName={storedThemeName}
      setThemeName={setThemeName}
    />
  );
};

export default ThemeProviderLoader;
