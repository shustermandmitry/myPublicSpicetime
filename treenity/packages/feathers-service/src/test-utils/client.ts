import {
  Application,
  defaultEventMap,
  defaultServiceMethods,
  RealTimeConnection,
  TransportConnection,
} from '@feathersjs/feathers';
import { Service, SocketService } from '@feathersjs/transport-commons/client';
import { runMethod } from '@feathersjs/transport-commons/lib/socket/utils';

export { SocketService };

export default function testClient<Services = any>(options?: any) {
  const defaultService = function (this: Application<any>, name: string) {
    const self = this;
    const events = Object.values(defaultEventMap);
    const connection = {
      emit(method: string, ...args: any[]) {
        return runMethod(self, {} as RealTimeConnection, name, method, args);
      },
    };

    const settings = Object.assign({}, options, {
      events,
      name,
      connection,
      method: 'emit',
    });

    return new Service(settings) as any;
  };

  const initialize = function (app: Application<Services>) {
    app.defaultService = defaultService;
    app.mixins.unshift((service, _location, options) => {
      if (options && options.methods && service instanceof Service) {
        const customMethods = options.methods.filter(name => !defaultServiceMethods.includes(name));

        service.methods(...customMethods);
      }
    });
  };

  initialize.Service = Service;
  initialize.service = defaultService;

  return initialize as TransportConnection<Services>;
}

if (typeof module !== 'undefined') {
  module.exports = Object.assign(testClient, module.exports);
}
