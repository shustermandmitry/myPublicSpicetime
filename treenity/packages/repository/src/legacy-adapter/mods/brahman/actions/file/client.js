import './TelegramActionFileSend.client';
import './TelegramActionFileUpload.client';
import './TelegramFile.client';
