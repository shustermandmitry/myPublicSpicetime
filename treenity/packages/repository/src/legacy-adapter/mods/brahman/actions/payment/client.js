import './TelegramActionGetWallet.client';
import './TelegramActionSendCoins.client';
import './TelegramActionSplitCoins.client';
