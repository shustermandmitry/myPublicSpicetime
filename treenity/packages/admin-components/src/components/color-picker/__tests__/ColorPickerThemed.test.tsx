import ColorPicker from '@/components/color-picker';
import ThemeProvider from '@/utils/ThemeProvider';
import { fireEvent, render, waitFor } from '@testing-library/react';
import { createCookies, parseCookies, setCookie, CookieContext } from '@treenity/repository/client';
// @ts-ignore
import { AggregationColor } from 'antd/lib/color-picker/color.js';
import React from 'react';

const cookies = createCookies(parseCookies(document.cookie), setCookie);

describe('ColorPicker', () => {
  const mockOnChange = jest.fn();
  const mockOnChangeComplete = jest.fn();

  beforeEach(() => {
    mockOnChange.mockClear();
    mockOnChangeComplete.mockClear();
  });

  it('renders color-picker component', () => {
    const { container } = render(
      <CookieContext.Provider value={cookies}>
        <ThemeProvider>
          <ColorPicker onChange={mockOnChange} />
        </ThemeProvider>
      </CookieContext.Provider>,
    );

    expect(container).toMatchSnapshot();
    const root = container.querySelector('div.css-pyqgzr');
    expect(root).toBeInTheDocument();

    const colorBlock = container.querySelector('.ant-color-picker-color-block');
    expect(colorBlock).toBeInTheDocument();

    const colorText = container.querySelector('.ant-color-picker-trigger-text');
    expect(colorText).toHaveTextContent('#27AE60');
  });

  it('opens color picker dropdown when clicked', async () => {
    const { container } = render(
      <CookieContext.Provider value={ cookies }>
        <ThemeProvider>
          <ColorPicker onChange={mockOnChange} />
        </ThemeProvider>
      </CookieContext.Provider>,
    );

    expect(container).toMatchSnapshot();
    const colorPicker = container.querySelector('.ant-color-picker-trigger');
    expect(colorPicker).toBeInTheDocument();
    fireEvent.click(colorPicker!);

    await waitFor(() => {
      const popover = container.querySelector('.ant-popover');
      expect(popover).toBeInTheDocument();
    });
  });

  it('changes color when a new color is selected', async () => {
    const { container } = render(
      <CookieContext.Provider value={ cookies }>
        <ThemeProvider>
          <ColorPicker onChange={mockOnChange} onChangeComplete={mockOnChangeComplete} />
        </ThemeProvider>
      </CookieContext.Provider>,
    );

    expect(container).toMatchSnapshot();
    const colorPicker = container.querySelector('.ant-color-picker-trigger');
    expect(colorPicker).toBeInTheDocument();
    fireEvent.click(colorPicker!);

    await waitFor(() => {
      const hexInput = container.querySelector('.ant-color-picker-hex-input input');
      expect(hexInput).toBeInTheDocument();
      fireEvent.change(hexInput!, { target: { value: 'ff0000' } });
    });

    expect(mockOnChange).toHaveBeenCalledWith(expect.any(AggregationColor), '#ff0000');
    expect(mockOnChangeComplete).toHaveBeenCalledWith(expect.any(AggregationColor));
  });

  it('resets to default color when reset button is clicked', async () => {
    const { container } = render(
      <CookieContext.Provider value={ cookies }>
        <ThemeProvider>
          <ColorPicker onChange={mockOnChange} onChangeComplete={mockOnChangeComplete} />
        </ThemeProvider>
      </CookieContext.Provider>,
    );

    expect(container).toMatchSnapshot();
    const resetButton = container.querySelector('button.ant-btn-text');
    expect(resetButton).toBeInTheDocument();
    fireEvent.click(resetButton!);

    await waitFor(() => {
      expect(mockOnChange).toHaveBeenCalledWith(
        expect.objectContaining({
          metaColor: expect.objectContaining({
            r: 39,
            g: 174,
            b: 96,
            a: 1,
          }),
        }),
        '27ae60',
      );
      expect(mockOnChangeComplete).toHaveBeenCalledWith(
        expect.objectContaining({
          metaColor: expect.objectContaining({
            r: 39,
            g: 174,
            b: 96,
            a: 1,
          }),
        }),
      );
    });
  });

  it('adds color to presets when "+ Add to Presets" is clicked', async () => {
    const { container } = render(
      <CookieContext.Provider value={ cookies }>
        <ThemeProvider>
          <ColorPicker onChange={mockOnChange} />
        </ThemeProvider>
      </CookieContext.Provider>,
    );

    expect(container).toMatchSnapshot();
    const colorPicker = container.querySelector('.ant-color-picker-trigger');
    expect(colorPicker).toBeInTheDocument();
    fireEvent.click(colorPicker!);

    await waitFor(() => {
      const addToPresetsButton = container.querySelector('.css-psta8o .css-1ro2kyp');
      expect(addToPresetsButton).toBeInTheDocument();
      fireEvent.click(addToPresetsButton!);
    });

    const presetColors = container.querySelectorAll('.css-psta8o .css-1ro2kyp');
    expect(presetColors.length).toBeGreaterThan(1);
  });

  it('selects color from presets when clicked', async () => {
    const { container } = render(
      <CookieContext.Provider value={ cookies }>
        <ThemeProvider>
          <ColorPicker onChange={mockOnChange} value="#27ae60" />
        </ThemeProvider>
      </CookieContext.Provider>,
    );

    expect(container).toMatchSnapshot();
    const colorPicker = container.querySelector('.ant-color-picker-trigger');
    expect(colorPicker).toBeInTheDocument();
    fireEvent.click(colorPicker!);

    await waitFor(() => {
      const popover = container.querySelector('.ant-popover');
      expect(popover).toBeInTheDocument();
    });

    const addToPresetsButton = container.querySelector('.css-psta8o .css-1ro2kyp');
    expect(addToPresetsButton).toBeInTheDocument();
    fireEvent.click(addToPresetsButton!);

    const presetColors = container.querySelector('.css-psta8o .css-14x6vl2');
    expect(presetColors).toBeInTheDocument();
    fireEvent.click(presetColors!);

    await waitFor(() => {
      expect(mockOnChange).toHaveBeenCalled();
    });

    const lastCall = mockOnChange.mock.calls[mockOnChange.mock.calls.length - 1];
    expect(lastCall[1]).toMatch(/^#[0-9a-f]{6}$/);
  });
});
