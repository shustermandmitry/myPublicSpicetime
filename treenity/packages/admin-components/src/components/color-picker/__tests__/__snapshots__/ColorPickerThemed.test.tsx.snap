// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker adds color to presets when "+ Add to Presets" is clicked 1`] = `
.emotion-0 {
  position: relative;
  line-height: 1px;
}

.emotion-0.emotion-0.emotion-0 .ant-collapse-header {
  display: none;
}

.emotion-0.emotion-0.emotion-0 .ant-color-picker-color-block.ant-color-picker-presets-color:before {
  border-radius: 6px;
}

.emotion-0 .ant-color-picker .ant-color-picker-inner .ant-color-picker-presets .ant-collapse-item>.ant-collapse-content>.ant-collapse-content-box {
  padding: 0;
}

.emotion-1 {
  width: 100%;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  padding-left: 6px;
  background-color: #fff;
  box-shadow: none!important;
  height: px;
  border-radius: px;
}

.emotion-1 .ant-color-picker-trigger-text {
  font-size: px;
}

.emotion-1 .ant-color-picker-color-block {
  width: 12px!important;
  height: 12px!important;
}

.emotion-4 {
  box-shadow: none;
  line-height: 100%;
  letter-spacing: -0.04em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-box-pack: none;
  -ms-flex-pack: none;
  -webkit-justify-content: none;
  justify-content: none;
  color: #27ae60;
  background: transparent!important;
  position: absolute;
  top: calc(50% - 10px);
  right: 0;
  width: 20px;
  height: 20px;
  padding: 2px;
  border: 0;
  color: rgba(38, 35, 63, 0.88);
}

.emotion-4>span {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -ms-flex-pack: space-around;
  -webkit-justify-content: space-around;
  justify-content: space-around;
}

.emotion-4.ant-btn.ant-btn-round {
  padding-inline: 12px;
}

.emotion-4 i {
  font-size: 16px;
}

.emotion-4>span {
  gap: 6px;
}

.emotion-4 span {
  font-size: 12px;
}

.emotion-4.emotion-4>span>span>i {
  font-size: 16px;
}

.emotion-4:hover {
  color: #ababab!important;
}

.emotion-4:active {
  color: !important;
}

.emotion-4:hover {
  color: #27ae60!important;
}

.emotion-4.emotion-4 i {
  font-size: 12px;
}

.emotion-5 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  text-align: center;
}

.emotion-6 [class^='icon-'],
.emotion-6 [class*=' icon-'] {
  font-size: 15px;
}

.emotion-7 {
  -webkit-transition: -webkit-transform 0.3s cubic-bezier(0.78, 0.14, 0.15, 0.86);
  transition: transform 0.3s cubic-bezier(0.78, 0.14, 0.15, 0.86);
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="ant-color-picker-trigger emotion-1 emotion-2"
    >
      <div
        class="ant-color-picker-color-block"
      >
        <div
          class="ant-color-picker-color-block-inner"
          style="background: rgb(39, 174, 96);"
        />
      </div>
      <div
        class="ant-color-picker-trigger-text"
      >
        #27AE60
      </div>
    </div>
    <button
      class="ant-btn emotion-2 ant-btn-text emotion-4"
      type="button"
    >
      <span>
        <span
          class="emotion-5"
        >
          <span
            class="anticon undefined emotion-6"
          >
            <i
              class="icon-close_outlined emotion-7"
            />
          </span>
        </span>
      </span>
    </button>
  </div>
</div>
`;

exports[`ColorPicker changes color when a new color is selected 1`] = `
.emotion-0 {
  position: relative;
  line-height: 1px;
}

.emotion-0.emotion-0.emotion-0 .ant-collapse-header {
  display: none;
}

.emotion-0.emotion-0.emotion-0 .ant-color-picker-color-block.ant-color-picker-presets-color:before {
  border-radius: 6px;
}

.emotion-0 .ant-color-picker .ant-color-picker-inner .ant-color-picker-presets .ant-collapse-item>.ant-collapse-content>.ant-collapse-content-box {
  padding: 0;
}

.emotion-1 {
  width: 100%;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  padding-left: 6px;
  background-color: #fff;
  box-shadow: none!important;
  height: px;
  border-radius: px;
}

.emotion-1 .ant-color-picker-trigger-text {
  font-size: px;
}

.emotion-1 .ant-color-picker-color-block {
  width: 12px!important;
  height: 12px!important;
}

.emotion-4 {
  box-shadow: none;
  line-height: 100%;
  letter-spacing: -0.04em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-box-pack: none;
  -ms-flex-pack: none;
  -webkit-justify-content: none;
  justify-content: none;
  color: #27ae60;
  background: transparent!important;
  position: absolute;
  top: calc(50% - 10px);
  right: 0;
  width: 20px;
  height: 20px;
  padding: 2px;
  border: 0;
  color: rgba(38, 35, 63, 0.88);
}

.emotion-4>span {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -ms-flex-pack: space-around;
  -webkit-justify-content: space-around;
  justify-content: space-around;
}

.emotion-4.ant-btn.ant-btn-round {
  padding-inline: 12px;
}

.emotion-4 i {
  font-size: 16px;
}

.emotion-4>span {
  gap: 6px;
}

.emotion-4 span {
  font-size: 12px;
}

.emotion-4.emotion-4>span>span>i {
  font-size: 16px;
}

.emotion-4:hover {
  color: #ababab!important;
}

.emotion-4:active {
  color: !important;
}

.emotion-4:hover {
  color: #27ae60!important;
}

.emotion-4.emotion-4 i {
  font-size: 12px;
}

.emotion-5 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  text-align: center;
}

.emotion-6 [class^='icon-'],
.emotion-6 [class*=' icon-'] {
  font-size: 15px;
}

.emotion-7 {
  -webkit-transition: -webkit-transform 0.3s cubic-bezier(0.78, 0.14, 0.15, 0.86);
  transition: transform 0.3s cubic-bezier(0.78, 0.14, 0.15, 0.86);
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="ant-color-picker-trigger emotion-1 emotion-2"
    >
      <div
        class="ant-color-picker-color-block"
      >
        <div
          class="ant-color-picker-color-block-inner"
          style="background: rgb(39, 174, 96);"
        />
      </div>
      <div
        class="ant-color-picker-trigger-text"
      >
        #27AE60
      </div>
    </div>
    <button
      class="ant-btn emotion-2 ant-btn-text emotion-4"
      type="button"
    >
      <span>
        <span
          class="emotion-5"
        >
          <span
            class="anticon undefined emotion-6"
          >
            <i
              class="icon-close_outlined emotion-7"
            />
          </span>
        </span>
      </span>
    </button>
  </div>
</div>
`;

exports[`ColorPicker opens color picker dropdown when clicked 1`] = `
.emotion-0 {
  position: relative;
  line-height: 1px;
}

.emotion-0.emotion-0.emotion-0 .ant-collapse-header {
  display: none;
}

.emotion-0.emotion-0.emotion-0 .ant-color-picker-color-block.ant-color-picker-presets-color:before {
  border-radius: 6px;
}

.emotion-0 .ant-color-picker .ant-color-picker-inner .ant-color-picker-presets .ant-collapse-item>.ant-collapse-content>.ant-collapse-content-box {
  padding: 0;
}

.emotion-1 {
  width: 100%;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  padding-left: 6px;
  background-color: #fff;
  box-shadow: none!important;
  height: px;
  border-radius: px;
}

.emotion-1 .ant-color-picker-trigger-text {
  font-size: px;
}

.emotion-1 .ant-color-picker-color-block {
  width: 12px!important;
  height: 12px!important;
}

.emotion-4 {
  box-shadow: none;
  line-height: 100%;
  letter-spacing: -0.04em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-box-pack: none;
  -ms-flex-pack: none;
  -webkit-justify-content: none;
  justify-content: none;
  color: #27ae60;
  background: transparent!important;
  position: absolute;
  top: calc(50% - 10px);
  right: 0;
  width: 20px;
  height: 20px;
  padding: 2px;
  border: 0;
  color: rgba(38, 35, 63, 0.88);
}

.emotion-4>span {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -ms-flex-pack: space-around;
  -webkit-justify-content: space-around;
  justify-content: space-around;
}

.emotion-4.ant-btn.ant-btn-round {
  padding-inline: 12px;
}

.emotion-4 i {
  font-size: 16px;
}

.emotion-4>span {
  gap: 6px;
}

.emotion-4 span {
  font-size: 12px;
}

.emotion-4.emotion-4>span>span>i {
  font-size: 16px;
}

.emotion-4:hover {
  color: #ababab!important;
}

.emotion-4:active {
  color: !important;
}

.emotion-4:hover {
  color: #27ae60!important;
}

.emotion-4.emotion-4 i {
  font-size: 12px;
}

.emotion-5 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  text-align: center;
}

.emotion-6 [class^='icon-'],
.emotion-6 [class*=' icon-'] {
  font-size: 15px;
}

.emotion-7 {
  -webkit-transition: -webkit-transform 0.3s cubic-bezier(0.78, 0.14, 0.15, 0.86);
  transition: transform 0.3s cubic-bezier(0.78, 0.14, 0.15, 0.86);
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="ant-color-picker-trigger emotion-1 emotion-2"
    >
      <div
        class="ant-color-picker-color-block"
      >
        <div
          class="ant-color-picker-color-block-inner"
          style="background: rgb(39, 174, 96);"
        />
      </div>
      <div
        class="ant-color-picker-trigger-text"
      >
        #27AE60
      </div>
    </div>
    <button
      class="ant-btn emotion-2 ant-btn-text emotion-4"
      type="button"
    >
      <span>
        <span
          class="emotion-5"
        >
          <span
            class="anticon undefined emotion-6"
          >
            <i
              class="icon-close_outlined emotion-7"
            />
          </span>
        </span>
      </span>
    </button>
  </div>
</div>
`;

exports[`ColorPicker renders ColorPicker component 1`] = `
.emotion-0 {
  position: relative;
  line-height: 1px;
}

.emotion-0.emotion-0.emotion-0 .ant-collapse-header {
  display: none;
}

.emotion-0.emotion-0.emotion-0 .ant-color-picker-color-block.ant-color-picker-presets-color:before {
  border-radius: 6px;
}

.emotion-0 .ant-color-picker .ant-color-picker-inner .ant-color-picker-presets .ant-collapse-item>.ant-collapse-content>.ant-collapse-content-box {
  padding: 0;
}

.emotion-1 {
  width: 100%;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  padding-left: 6px;
  background-color: #fff;
  box-shadow: none!important;
  height: px;
  border-radius: px;
}

.emotion-1 .ant-color-picker-trigger-text {
  font-size: px;
}

.emotion-1 .ant-color-picker-color-block {
  width: 12px!important;
  height: 12px!important;
}

.emotion-4 {
  box-shadow: none;
  line-height: 100%;
  letter-spacing: -0.04em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-box-pack: none;
  -ms-flex-pack: none;
  -webkit-justify-content: none;
  justify-content: none;
  color: #27ae60;
  background: transparent!important;
  position: absolute;
  top: calc(50% - 10px);
  right: 0;
  width: 20px;
  height: 20px;
  padding: 2px;
  border: 0;
  color: rgba(38, 35, 63, 0.88);
}

.emotion-4>span {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -ms-flex-pack: space-around;
  -webkit-justify-content: space-around;
  justify-content: space-around;
}

.emotion-4.ant-btn.ant-btn-round {
  padding-inline: 12px;
}

.emotion-4 i {
  font-size: 16px;
}

.emotion-4>span {
  gap: 6px;
}

.emotion-4 span {
  font-size: 12px;
}

.emotion-4.emotion-4>span>span>i {
  font-size: 16px;
}

.emotion-4:hover {
  color: #ababab!important;
}

.emotion-4:active {
  color: !important;
}

.emotion-4:hover {
  color: #27ae60!important;
}

.emotion-4.emotion-4 i {
  font-size: 12px;
}

.emotion-5 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  text-align: center;
}

.emotion-6 [class^='icon-'],
.emotion-6 [class*=' icon-'] {
  font-size: 15px;
}

.emotion-7 {
  -webkit-transition: -webkit-transform 0.3s cubic-bezier(0.78, 0.14, 0.15, 0.86);
  transition: transform 0.3s cubic-bezier(0.78, 0.14, 0.15, 0.86);
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="ant-color-picker-trigger emotion-1 emotion-2"
    >
      <div
        class="ant-color-picker-color-block"
      >
        <div
          class="ant-color-picker-color-block-inner"
          style="background: rgb(39, 174, 96);"
        />
      </div>
      <div
        class="ant-color-picker-trigger-text"
      >
        #27AE60
      </div>
    </div>
    <button
      class="ant-btn emotion-2 ant-btn-text emotion-4"
      type="button"
    >
      <span>
        <span
          class="emotion-5"
        >
          <span
            class="anticon undefined emotion-6"
          >
            <i
              class="icon-close_outlined emotion-7"
            />
          </span>
        </span>
      </span>
    </button>
  </div>
</div>
`;

exports[`ColorPicker resets to default color when reset button is clicked 1`] = `
.emotion-0 {
  position: relative;
  line-height: 1px;
}

.emotion-0.emotion-0.emotion-0 .ant-collapse-header {
  display: none;
}

.emotion-0.emotion-0.emotion-0 .ant-color-picker-color-block.ant-color-picker-presets-color:before {
  border-radius: 6px;
}

.emotion-0 .ant-color-picker .ant-color-picker-inner .ant-color-picker-presets .ant-collapse-item>.ant-collapse-content>.ant-collapse-content-box {
  padding: 0;
}

.emotion-1 {
  width: 100%;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  padding-left: 6px;
  background-color: #fff;
  box-shadow: none!important;
  height: px;
  border-radius: px;
}

.emotion-1 .ant-color-picker-trigger-text {
  font-size: px;
}

.emotion-1 .ant-color-picker-color-block {
  width: 12px!important;
  height: 12px!important;
}

.emotion-4 {
  box-shadow: none;
  line-height: 100%;
  letter-spacing: -0.04em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-box-pack: none;
  -ms-flex-pack: none;
  -webkit-justify-content: none;
  justify-content: none;
  color: #27ae60;
  background: transparent!important;
  position: absolute;
  top: calc(50% - 10px);
  right: 0;
  width: 20px;
  height: 20px;
  padding: 2px;
  border: 0;
  color: rgba(38, 35, 63, 0.88);
}

.emotion-4>span {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -ms-flex-pack: space-around;
  -webkit-justify-content: space-around;
  justify-content: space-around;
}

.emotion-4.ant-btn.ant-btn-round {
  padding-inline: 12px;
}

.emotion-4 i {
  font-size: 16px;
}

.emotion-4>span {
  gap: 6px;
}

.emotion-4 span {
  font-size: 12px;
}

.emotion-4.emotion-4>span>span>i {
  font-size: 16px;
}

.emotion-4:hover {
  color: #ababab!important;
}

.emotion-4:active {
  color: !important;
}

.emotion-4:hover {
  color: #27ae60!important;
}

.emotion-4.emotion-4 i {
  font-size: 12px;
}

.emotion-5 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  text-align: center;
}

.emotion-6 [class^='icon-'],
.emotion-6 [class*=' icon-'] {
  font-size: 15px;
}

.emotion-7 {
  -webkit-transition: -webkit-transform 0.3s cubic-bezier(0.78, 0.14, 0.15, 0.86);
  transition: transform 0.3s cubic-bezier(0.78, 0.14, 0.15, 0.86);
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="ant-color-picker-trigger emotion-1 emotion-2"
    >
      <div
        class="ant-color-picker-color-block"
      >
        <div
          class="ant-color-picker-color-block-inner"
          style="background: rgb(39, 174, 96);"
        />
      </div>
      <div
        class="ant-color-picker-trigger-text"
      >
        #27AE60
      </div>
    </div>
    <button
      class="ant-btn emotion-2 ant-btn-text emotion-4"
      type="button"
    >
      <span>
        <span
          class="emotion-5"
        >
          <span
            class="anticon undefined emotion-6"
          >
            <i
              class="icon-close_outlined emotion-7"
            />
          </span>
        </span>
      </span>
    </button>
  </div>
</div>
`;

exports[`ColorPicker selects color from presets when clicked 1`] = `
.emotion-0 {
  position: relative;
  line-height: 1px;
}

.emotion-0.emotion-0.emotion-0 .ant-collapse-header {
  display: none;
}

.emotion-0.emotion-0.emotion-0 .ant-color-picker-color-block.ant-color-picker-presets-color:before {
  border-radius: 6px;
}

.emotion-0 .ant-color-picker .ant-color-picker-inner .ant-color-picker-presets .ant-collapse-item>.ant-collapse-content>.ant-collapse-content-box {
  padding: 0;
}

.emotion-1 {
  width: 100%;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  padding-left: 6px;
  background-color: #fff;
  box-shadow: none!important;
  height: px;
  border-radius: px;
}

.emotion-1 .ant-color-picker-trigger-text {
  font-size: px;
}

.emotion-1 .ant-color-picker-color-block {
  width: 12px!important;
  height: 12px!important;
}

.emotion-4 {
  box-shadow: none;
  line-height: 100%;
  letter-spacing: -0.04em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-box-pack: none;
  -ms-flex-pack: none;
  -webkit-justify-content: none;
  justify-content: none;
  color: #27ae60;
  background: transparent!important;
  position: absolute;
  top: calc(50% - 10px);
  right: 0;
  width: 20px;
  height: 20px;
  padding: 2px;
  border: 0;
  color: rgba(38, 35, 63, 0.88);
}

.emotion-4>span {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -ms-flex-pack: space-around;
  -webkit-justify-content: space-around;
  justify-content: space-around;
}

.emotion-4.ant-btn.ant-btn-round {
  padding-inline: 12px;
}

.emotion-4 i {
  font-size: 16px;
}

.emotion-4>span {
  gap: 6px;
}

.emotion-4 span {
  font-size: 12px;
}

.emotion-4.emotion-4>span>span>i {
  font-size: 16px;
}

.emotion-4:hover {
  color: #ababab!important;
}

.emotion-4:active {
  color: !important;
}

.emotion-4:hover {
  color: #27ae60!important;
}

.emotion-4.emotion-4 i {
  font-size: 12px;
}

.emotion-5 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  text-align: center;
}

.emotion-6 [class^='icon-'],
.emotion-6 [class*=' icon-'] {
  font-size: 15px;
}

.emotion-7 {
  -webkit-transition: -webkit-transform 0.3s cubic-bezier(0.78, 0.14, 0.15, 0.86);
  transition: transform 0.3s cubic-bezier(0.78, 0.14, 0.15, 0.86);
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="ant-color-picker-trigger emotion-1 emotion-2"
    >
      <div
        class="ant-color-picker-color-block"
      >
        <div
          class="ant-color-picker-color-block-inner"
          style="background: rgb(39, 174, 96);"
        />
      </div>
      <div
        class="ant-color-picker-trigger-text"
      >
        #27AE60
      </div>
    </div>
    <button
      class="ant-btn emotion-2 ant-btn-text emotion-4"
      type="button"
    >
      <span>
        <span
          class="emotion-5"
        >
          <span
            class="anticon undefined emotion-6"
          >
            <i
              class="icon-close_outlined emotion-7"
            />
          </span>
        </span>
      </span>
    </button>
  </div>
</div>
`;
